platform_specific/ubuntu/flash_driver.cpp

```diff
@@ -7,36 +7,60 @@
  */
 
 #include "flash_driver.h"
+
 #include <stdbool.h>
 #include <string.h>
-#include <iostream>
-#include <iomanip>
+
+#include <cassert>
+#include <filesystem>
 #include <fstream>
+#include <iomanip>
+#include <iostream>
 #include <string>
-#include <cassert>
+
+#include "YamlParameters.hpp"
 #include "libparams_error_codes.h"
+#include "params.hpp"
 #include "storage.h"
 
 #define PAGE_SIZE_BYTES 2048
 
-uint8_t flash_memory[PAGE_SIZE_BYTES];
+namespace fs = std::filesystem;
+extern IntegerDesc_t integer_desc_pool[];
+extern StringDesc_t string_desc_pool[];
+
+uint8_t *flash_memory;
```

Comment:
the size is know on the compilation time. why not to use static array with fixed size?